### PR TITLE
Overhaul error handling

### PR DESCRIPTION
**Pull request by Derpthemeus**
This addresses the backwards compatible parts of #54

### CODE REVIEW

File: routes/weather.ts [hunk 1]
```diff
@@ -259,13 +267,23 @@ export const getWateringData = async function( req: express.Request, res: expres
 			return;
 		}
 
-		wateringData = await weatherProvider.getWateringData( coordinates );
+		try {
+			wateringData = await weatherProvider.getWateringData( coordinates );
```
Comment (PeteBa) [inline routes/weather.ts]: `weatherProvider` base class should define default `getWateringData()` method to raise exception if not overloaded by subclass. That way we wouldn't need L265-269.

File: routes/weather.ts [hunk 2]
```diff
@@ -259,13 +267,23 @@ export const getWateringData = async function( req: express.Request, res: expres
...
-		scale = calculateZimmermanWateringScale( adjustmentOptions, wateringData );
+		try {
+			scale = calculateZimmermanWateringScale( adjustmentOptions, wateringData );
+		} catch ( err ) {
+			res.send( "Error: " + err );
+			return;
+		}
 	}
 
 	if (wateringData) {
```
Comment (PeteBa) [inline routes/weather.ts]: We shouldnt need this check given try..catch above.

Comment (Derpthemeus) [reply routes/weather.ts]: `wateringData` can still be `undefined` if the manual adjustment method is being used

File: routes/weather.ts (hunk 1 above)
Comment (Derpthemeus) [reply routes/weather.ts]: Resolved in b57b019

File: routes/weather.ts (hunk 2 above)
Comment (PeteBa) [reply routes/weather.ts]: You are right.

 I just don't like the logic in this area. It is too convoluted with nested ifs and `> METHOD.MANUAL` conditions that all feels a bit hard to decipher. Not one for this PR but I'll open an issue for consideration at a latter point.

**Pull request merged**